File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Unit/Monster/Model/TheBottom/Lagavulin_Model.ts
import { Mathf } from "../../../../../../../Core/Module/Math/Mathf";
import DungeonManager from "../../../../DungeonManager";
import FileHelper from "../../../../FileHelper";
import { ILoader } from "../../../../Loader/ILoader";
import { SpineLoader } from "../../../../Loader/SpineLoader";
import { MonsterInfo } from "../../MonsterInfo";

export class Lagavulin_Model extends MonsterInfo{
    public static Id = "Lagavulin"

    get MaxHealth(): number {
        if(DungeonManager.Inst.AdvanceLevel >= 7)
            return Mathf.RandomRange(112, 115);
        else
            return Mathf.RandomRange(109, 111);
    }
    get Loader(): ILoader {
        return new SpineLoader(FileHelper.FormatPath("Monsters/theBottom/lagavulin/skeleton.atlas"),FileHelper.FormatPath("Monsters/theBottom/lagavulin/skeleton.json"),FileHelper.FormatPath("Monsters/theBottom/lagavulin/skeleton.png"));
    }

    get Name(): string {
        return "乐加维林";
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Unit/Monster/Model/TheBottom/Lagavulin_Model.cs
using System.Collections.Generic;
using ModTheSpire.Dungeon;
using ModTheSpire.Power;
using ModTheSpire.Unit.Loader;
using UnityEngine;

namespace ModTheSpire.Unit.Monster.Model.TheBottom
{
    public class Lagavulin_Model : MonsterInfo
    {
        public static string Id = "Lagavulin";
        private ILoader loader;

        public override int MaxHealth => DungeonManager.Inst.AdvanceLevel >= 7 ? Random.Range(112, 115) : Random.Range(109,111);

        public override ILoader Loader
        {
            get
            {
                return this.loader ??= new SpineLoader("Monsters/theBottom/lagavulin/skeleton.atlas",
                    "Monsters/theBottom/lagavulin/skeleton.json", "Monsters/theBottom/lagavulin/skeleton.png");
            }
        }

        public override string Name => "乐加维林";

        public Lagavulin_Model(Vector2 pos) : base(pos)
        {
        }
    }
}